examples/udev.rs

```diff
@@ -0,0 +1,472 @@
+            .filter(|conn| conn.connection_state() == ConnectorState::Connected)
+            .inspect(|conn| info!(self.logger, "Connected: {:?}", conn.connector_type()))
+            .collect();
+
+        let mut used_crtcs: HashSet<crtc::Handle> = HashSet::new();
+
+        let mut state = state.into();
+
+        // very naive way of finding good crtc/encoder/connector combinations. This problem is np-complete
+        for connector_info in connector_infos {
+            let encoder_infos = connector_info.encoders().iter().flat_map(|encoder_handle| EncoderInfo::load_from_device(device, *encoder_handle)).collect::<Vec<EncoderInfo>>();
+            for encoder_info in encoder_infos {
+                for crtc in res_handles.filter_crtcs(encoder_info.possible_crtcs()) {
+                    if !used_crtcs.contains(&crtc) {
+                        let mode = connector_info.modes()[0]; // Use first mode (usually highest resoltion, but in reality you should filter and sort and check and match with other connectors, if you use more then one.)
+                        // create a backend
+                        let renderer_token = device.create_backend(&mut state, crtc, mode, vec![connector_info.handle()]).unwrap();
+
+                        // create cursor
+                        {
+                            let renderer = state.get_mut(renderer_token);
+                            renderer.set_cursor_representation(&self.pointer_image, (2, 2)).unwrap();
+                        }
+
+                        // render first frame
+                        {
+                            let renderer = state.get_mut(renderer_token);
+                            let mut frame = renderer.draw();
+                            frame.clear_color(0.8, 0.8, 0.9, 1.0);
+                            frame.finish().unwrap();
+                        }
+
+                        used_crtcs.insert(crtc);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+}
+
+impl UdevHandler<GliumDrawer<DrmBackend>, DrmHandlerImpl> for UdevHandlerImpl {
+    fn device_added<'a, S: Into<StateProxy<'a>>>(&mut self, state: S, device: &mut DrmDevice<GliumDrawer<DrmBackend>>) -> Option<DrmHandlerImpl>
+    {
+        self.scan_connectors(state, device);
+
+        Some(DrmHandlerImpl {
+            shell_state_token: self.shell_state_token.clone(),
+            compositor_token: self.compositor_token.clone(),
+            window_map: self.window_map.clone(),
+            pointer_location: self.pointer_location.clone(),
+            logger: self.logger.clone(),
+        })
+    }
+
+    fn device_changed<'a, S: Into<StateProxy<'a>>>(&mut self, state: S, device: &StateToken<DrmDevice<GliumDrawer<DrmBackend>>>) {
+        //quick and dirty
+        let mut state = state.into();
+        self.device_removed(&mut state, device);
+        state.with_value(device, |state, device| self.scan_connectors(state, device));
```

Comment:
So this goes the "quick & dirty way" and removes then re-init the device, right? (nothing wring with that, just checking I'm understanding correctly)